@maeste This is failing CI due to a checkstyle problem:

[10:36:31][Step 2/4] [INFO] --- maven-checkstyle-plugin:3.0.0:checkstyle (check-style) @ wildfly-connector ---
[10:36:32][Step 2/4] [INFO] Starting audit...
[10:36:32][Step 2/4] [ERROR] /store/work/tc-work/bb15431f347cd651/connector/src/main/java/org/jboss/as/connector/services/resourceadapters/deployment/AbstractResourceAdapterDeploymentService.java:31:8: Unused import - javax.security.auth.Subject. [UnusedImports]
[10:36:32][Step 2/4] Audit done.
[10:36:32][Step 2/4] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:checkstyle (check-style) on project wildfly-connector: An error has occurred in Checkstyle report generation.

--- a/connector/src/main/java/org/jboss/as/connector/services/resourceadapters/deployment/AbstractResourceAdapterDeploymentService.java
+++ b/connector/src/main/java/org/jboss/as/connector/services/resourceadapters/deployment/AbstractResourceAdapterDeploymentService.java
@@ -26,9 +26,9 @@ import static java.lang.Thread.currentThread;
 import static java.security.AccessController.doPrivileged;
 import static org.jboss.as.connector.logging.ConnectorLogger.DEPLOYMENT_CONNECTOR_LOGGER;
 
+import javax.naming.InitialContext;
 import javax.naming.Reference;
 import javax.resource.spi.ResourceAdapter;
-import javax.security.auth.Subject;
 import javax.transaction.TransactionManager;
 import java.io.File;
 import java.io.PrintWriter;
@@ -42,7 +42,6 @@ import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ThreadFactory;
-import javax.naming.InitialContext;
 
 import org.jboss.as.connector.logging.ConnectorLogger;
 import org.jboss.as.connector.metadata.api.resourceadapter.WorkManagerSecurity;
@@ -683,23 +682,7 @@ public abstract class AbstractResourceAdapterDeploymentService {
             } else if (securityDomain == null || securityDomain.trim().equals("")) {
                 return null;
             } else {
-                return new PicketBoxSubjectFactory(subjectFactory.getValue()){
-
-                    @Override
-                    public Subject createSubject(final String sd) {
-                        ServerSecurityManager sm = secManager.getOptionalValue();
-                        if (sm != null) {
-                            sm.push(sd);
-                        }
-                        try {
-                            return super.createSubject(sd);
-                        } finally {
-                            if (sm != null) {
-                                sm.pop();
-                            }
-                        }
-                    }
-                };
+                return new PicketBoxSubjectFactory(subjectFactory.getValue());
             }
         }
 

